### PR TITLE
feat: add existing region middleware tests

### PR DESCRIPTION
**Pull request by leonlafa**
## What type of PR is this? (check all applicable)

- [ ] 💡 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test

## Fixes issue
Fixes[ #157](https://github.com/Energy-Price-News-API/energy-prices-api/issues/157)

## Screenshots (if appropriate)
N/A

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by leonlafa** [comment]
@MizouziE

File: __tests__/unit-tests/existingRegionMiddleware.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    res.status = jest.fn().mockReturnValue(res);
+    res.sendFile =  jest.fn()
+
+    it.each([
+        [{params: {regionId: "asia"}}],
+        [{params: {regionId: "americas"}}],
+        [{params: {regionId: "europe"}}],
+    ])('should pass control to the next middleware when a valid region Id has been sent in the request', (req) => {
+        existingRegionMiddleware(req, res, next);
+        expect(next).toHaveBeenCalled()
```
Comment (MizouziE) [inline __tests__/unit-tests/existingRegionMiddleware.test.js:17]: Nice! That has just taught me a whole new spectrum of testing possibilities with this testing library. Thank you!

File: __tests__/unit-tests/existingRegionMiddleware.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    it.each([
+        [{params: {regionId: "africa"}}],
+        [{params: {regionId: "australia & oceania"}}],
+        [{params: {regionId: ""}}],
```
Comment (MizouziE) [inline __tests__/unit-tests/existingRegionMiddleware.test.js:23]: I will need to make a mental note to advise anyone adding sources from these regions to alter this test. Very good call though.

**Review by MizouziE — APPROVED** [review]
Cool tests, thank you! The `package.json` shuffle is the first time I've seen the dependencies just re-ordered on a PR :joy: but it causes no harm, so why not?

File: __tests__/unit-tests/existingRegionMiddleware.test.js (hunk 1 above)
Comment (leonlafa) [reply __tests__/unit-tests/existingRegionMiddleware.test.js:17]: You're welcome 😁.  I'll jump on some more testing shortly.

**Comment by leonlafa** [comment]
> Cool tests, thank you! The `package.json` shuffle is the first time I've seen the dependencies just re-ordered on a PR 😂 but it causes no harm, so why not?

Yeah, I thought that was weird. I'm not sure why that happened.  I was anticipating you would mention something!  😁

**Pull request merged**